### PR TITLE
fix: Properly handle file paths for Windows

### PR DESCRIPTION
**Pull request by bbernays**
#### Summary

closes: #14105 

Tested that this works properly on both MacOS and Windows

### CODE REVIEW

File: plugins/destination/snowflake/client/write.go [hunk 1]
```diff
@@ -54,11 +55,13 @@ func (c *Client) WriteTableBatch(ctx context.Context, name string, msgs message.
 	if err := f.Close(); err != nil {
 		return fmt.Errorf("failed to close temp file with last resource %s: %w", f.Name(), err)
 	}
-	sql := fmt.Sprintf(putFileIntoStage, f.Name())
+
+	sql := fmt.Sprintf(putFileIntoStage, strings.ReplaceAll(f.Name(), "\\", "\\\\"))
```
Comment (disq) [inline plugins/destination/snowflake/client/write.go:59]: Could use `strconv.Quote` or refactor into its own little fn...

**Review by disq — APPROVED** [review]

File: plugins/destination/snowflake/client/write.go [hunk 2]
```diff
@@ -54,13 +55,20 @@ func (c *Client) WriteTableBatch(ctx context.Context, name string, msgs message.
...
+	sql = fmt.Sprintf(copyIntoTable, tableName, escapePath(path.Base(f.Name())))
 	if _, err := c.db.ExecContext(ctx, sql); err != nil {
 		return fmt.Errorf("failed to copy file into table with last resource %s: %w", sql, err)
 	}
 	return err
 }
+
+// escapePath properly escapes the `\` character in window's file paths.
+func escapePath(p string) string {
+	return strings.ReplaceAll(p, "\\", "\\\\")
```
Comment (mnorbury) [inline plugins/destination/snowflake/client/write.go:73]: Is escaping the backslash the same as converting to a forward slash - which the docs recommended?

Comment (bbernays) [reply plugins/destination/snowflake/client/write.go:73]: The solution actually comes from the Snowflake driver docs:
https://pkg.go.dev/github.com/snowflakedb/gosnowflake#hdr-Support_For_PUT_and_GET

File: plugins/destination/snowflake/client/write.go [hunk 3]
```diff
@@ -54,13 +55,20 @@ func (c *Client) WriteTableBatch(ctx context.Context, name string, msgs message.
...
-	sql = fmt.Sprintf(copyIntoTable, tableName, path.Base(f.Name()))
+	sql = fmt.Sprintf(copyIntoTable, tableName, escapePath(path.Base(f.Name())))
 	if _, err := c.db.ExecContext(ctx, sql); err != nil {
 		return fmt.Errorf("failed to copy file into table with last resource %s: %w", sql, err)
 	}
 	return err
 }
+
+// escapePath properly escapes the `\` character in window's file paths.
+func escapePath(p string) string {
```
Comment (candiduslynx) [inline plugins/destination/snowflake/client/write.go:72]: why not use https://pkg.go.dev/path/filepath package & funcs from it?

Comment (candiduslynx) [reply plugins/destination/snowflake/client/write.go:72]: Specifically, they do handle this: https://cs.opensource.google/go/go/+/refs/tags/go1.21.1:src/path/filepath/path_windows.go

Comment (bbernays) [reply plugins/destination/snowflake/client/write.go:72]: What function do you want me to use?

Comment (candiduslynx) [reply plugins/destination/snowflake/client/write.go:72]: Oh, I missed that it is escape, not unescape

**Pull request merged**